### PR TITLE
Added code for fine-tuning DeepSeek

### PR DESCRIPTION
**Pull request by ahmed-boutar**
Big commit here, but don't worry it's not that complicated 
## Changes 
- Finished generating all the training data (~140 of the files changed represent the training data for about 140 episodes of Huberman Lab)
- Created a new directory ``deep_learning_model``, which contains the script to fine-tune the DeepSeek 1.3B model. Tried the 7B but it crashed my computer like 10 times with different parameters (tried to reduce batch size and increase the gradient accumulation step but nothing helped). Couldn't train the 1.3B since I have to use the cpu (estimated time for one epoch was about 6 days). I just copy pasted that code into Colab and ran the training there with the same exact script. Note the new directory ``deep_learning_model/fine_tuned_models`` was used for checkpoints when I tried to fine-tune locally so we can just delete it in the next commit. 
- Updated requirements.txt 

Tried to organize the repo by adding the folder but it didn't help much lol. Will get to it once we have the server code in AWS set up and the full project done since it won't take up that much time.

### CODE REVIEW

**Comment by notthattal** [comment]
Now that it's committed we can just connect a colab directly to this repo and run it on an A100 or T4. That way we all have access to it in case you your runtime disconnects again.

**Review by notthattal — APPROVED** [review]
